code/game/objects/items/weapons/lighters.dm

```diff
@@ -302,3 +312,31 @@
 /obj/item/match/firebrand/New()
 	..()
 	matchignite()
+
+
+/obj/item/match/unathi
+	name = "small blaze"
+	desc = "A little flame of your own, currently located dangerously in your mouth."
+	icon_state = "match_unathi"
+	force = 0
+	flags = DROPDEL
+	origin_tech = null
+	w_class = WEIGHT_CLASS_BULKY //to prevent it going to pockets
+
+/obj/item/match/unathi/New()
+	..()
+	matchignite()
+	name = "small blaze"
+	desc = "A little flame of your own, currently located dangerously in your mouth."
+	icon_state = "match_unathi"
+	force = 0
+	attack_verb = null
+
+/obj/item/match/unathi/matchburnout()
+	if(lit)
+		lit = FALSE
+		burnt = TRUE
```

Comment:
These seem unnecessary if you're deleting the fire right after?
Also, styling nitpick, you could switch the `if(lit)` check into a guard clause, then unindent the rest

Reply:
`lit = FALSE` is needed to stop a qdel loop